tests/DatastreamTest.php

```diff
@@ -17,6 +17,8 @@ protected function setUp() {
     $this->api = new FedoraApi($connection);
     $cache = new SimpleCache();
     $this->repository = new FedoraRepository($this->api, $cache);
+    $describe = $this->api->a->describeRepository();
+    $this->fedoraVersion = isset($describe['repositoryVersion']) ? $describe['repositoryVersion'] : NULL;
```

Comment:
Almost... feels like there should be some form of notice thrown, if we set a ```NULL``` here.

Reply:
Don't think there is a mechanism within PHPUnit to do this kind of messaging. Think we are ok to go with a `NULL` for now.

Reply:
@jordan: so if fedora version is not returned from API-A we are setting it to NULL, but that really means fedora is not working at all, so nothing else that depends on fedora being there run. So, should we either `throw new Exception()` even when that will really just set also $this->fedoraVersion to null... or  use [`::markTestSkipped`](https://phpunit.de/manual/6.5/en/incomplete-and-skipped-tests.html) with a "The Fedora API is not responding" message or something (never have done that before but a google search shows its an alternative)? @adam-vessey?

Reply:
@DiegoPino: ... I'm not too concerned... the tests should explode in different, exciting ways if Fedora is non-responsive. A bit of discussion w/ Jordan, the `isset()` was a naive attempt to allow the tests to run against alternative implementations of the Fedora 3 REST API (such as our [dgi_repo](https://github.com/discoverygarden/dgi_repo) implementation); however, [our dgi_repo implementations _does_ provide a version in its response](https://github.com/discoverygarden/dgi_repo/blob/1bb30d51a4cefc2ae510f5aeac4b36ed7a78f6a6/dgi_repo/fcrepo3/api.py#L130)... and I'm not aware of any other implementations... shrug?

You pinged the wrong Jordan (`@jordandukart`, not `@jordan`), by the way...